semgrep/semgrep/semgrep_main.py

```diff
@@ -61,8 +61,18 @@ def validate_single_rule(
             + [e.span for e in extra_key_spans],
             help=help_msg,
         )
-
-    # Raises InvalidRuleSchemaError if fails to parse
+    else:
```

Comment:
I'm guessing we can probably simplify this logic with the above. Happy to hack through it with you.